test/unit/federation_server_test.js

```diff
@@ -227,5 +227,98 @@ FEDERATION_SERVER="https://api.stellar.org/federation"
+
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 10000);
+      }).listen(4444, () => {
+        new StellarSdk.FederationServer('http://localhost:4444/federation', 'stellar.org', {allowHttp: true})
+          .resolveAddress('bob*stellar.org')
+          .should.be.rejectedWith(/timeout of 100ms exceeded/)
+          .notify(done)
+          .then(() => tempServer.close());
+      });
+    });
+
+    it("resolveAccountId times out", function (done) {
+      StellarSdk.Config.setTimeout(100);
+
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 10000);
+      }).listen(4444, () => {
+        new StellarSdk.FederationServer('http://localhost:4444/federation', 'stellar.org', {allowHttp: true})
+          .resolveAccountId('GB5XVAABEQMY63WTHDQ5RXADGYF345VWMNPTN2GFUDZT57D57ZQTJ7PS')
+          .should.be.rejectedWith(/timeout of 100ms exceeded/)
+          .notify(done)
+          .then(() => tempServer.close());
+      });
+    });
+
+    it("resolveTransactionId times out", function (done) {
+      StellarSdk.Config.setTimeout(100);
+
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 10000);
+      }).listen(4444, () => {
+        new StellarSdk.FederationServer('http://localhost:4444/federation', 'stellar.org', {allowHttp: true})
+          .resolveTransactionId('3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889')
+          .should.be.rejectedWith(/timeout of 100ms exceeded/)
+          .notify(done)
+          .then(() => tempServer.close());
+      });
+    });
+
+    it("createForDomain times out", function (done) {
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 1000);
+      }).listen(4444, () => {
+        StellarSdk.FederationServer.createForDomain("localhost:4444", {allowHttp: true, timeout: 100})
+          .should.be.rejectedWith(/timeout of 100ms exceeded/)
+          .notify(done)
+          .then(() => tempServer.close());
+      });
+    });
+
+    it("resolve times out", function (done) {
+      // Unable to create temp server in a browser
+      if (typeof window != 'undefined') {
+        return done();
+      }
+
+      let tempServer = http.createServer((req, res) => {
+        setTimeout(() => {}, 10000);
+      }).listen(4444, () => {
+        StellarSdk.FederationServer
+          .resolve('bob*localhost:4444', {allowHttp: true, timeout: 100})
```

Comment:
This test shouldn't work because `opts` is not passed to `FederationServer.constructor` (see other comment). It only works because `StellarSdk.Config.setTimeout(100);` is not reset after previous tests.